### PR TITLE
Add personas and user story documents

### PR DESCRIPTION
**Pull request by JAORMX**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

This adds persona definitions and user stories for the operator.
These came from our original discussion in a document [1], and are meant
to represent the actual people that this operator would help (the
personas) and the ways that the operator would help these people (as
user stories).

Thus, for future features added to this operator, they should be aimed
at helping one of the personas listed, or at improving or addressing one
of the user stories listed. New personas and user stories can be added,
and doing so will help clarify new functionality proposed to the
operator.

A section to the GitHub templates has been added to include references
to the user stories and personas.

[1] https://docs.google.com/document/d/1x0kS1Mzxcw0k3aq_27QK0OBHSea6apD-WMVwURYZsU4/edit?usp=sharing

#### Does this PR have test?

N/A.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

Co-Authored-By: @pjbgf 
Co-Authored-By: @saschagrunert 
Co-Authored-By: @cmurphy 
Co-Authored-By: @jhrozek

### CODE REVIEW

**Comment by JAORMX** [comment]
/hold

Need review from @cmurphy @pjbgf @saschagrunert and @jhrozek

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (saschagrunert) [inline .github/PULL_REQUEST_TEMPLATE.md:65]: Not sure if we need to add it to the PR template.

File: doc/personas.md
Comment (saschagrunert) [inline doc/personas.md:2]: (non blocking) Just a personal favor but I'd prefer the markdown style with fewer characters:
```suggestion
# Personas
```

**Review by saschagrunert — COMMENTED** [review]
Thank you for the work you put into this @JAORMX! :pray: 

I'd like to see it linked from the README.md, too :+1:

File: doc/personas.md
Comment (JAORMX) [reply doc/personas.md]: done

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (JAORMX) [reply .github/PULL_REQUEST_TEMPLATE.md:65]: I thought it would be a good idea since it might encourage people to revisit the user stories and thus help them re-think how their code is meant to help folks. Let's get more input on this. If more folks aren't keen on adding it to the PR template, we can remove it.

**Comment by JAORMX** [comment]
> Thank you for the work you put into this @JAORMX! pray

Thank you all for taking the time to go through them!

> 
> I'd like to see it linked from the README.md, too +1

Done

**Comment by jhrozek** [comment]
In general the text reads well. I only wonder about the open questions, because I wouldn't like us to just bury then in a document and forget. Should we file tickets about them now or should we bring them up again on the next team meeting?

**Comment by jhrozek** [comment]
> In general the text reads well. I only wonder about the open questions, because I wouldn't like us to just bury then in a document and forget. Should we file tickets about them now or should we bring them up again on the next team meeting?

btw I can help with either of those, didn't mean to pile even more work on you :-)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (pjbgf) [reply .github/PULL_REQUEST_TEMPLATE.md:65]: That's an interesting point. Assuming that meaningful PRs would always be linked to an issue, this extra link to user stories may be redundant.

Thinking out loud, Issues generally defines the `what` and `why` of a change - the latter point now being linked to an user story. Now the PRs  defines the `how`, which should be linked to an issue giving us already the full chain. But happy to hear other views.

**Review by pjbgf — COMMENTED** [review]
Really awesome stuff! I think this captures very well the personas and their user stories and should help new contributors understand better the problems we are trying to solve. :+1:  

The only thing that I would potentially change is the PR template as commented below.

**Comment by JAORMX** [comment]
> In general the text reads well. I only wonder about the open questions, because I wouldn't like us to just bury then in a document and forget. Should we file tickets about them now or should we bring them up again on the next team meeting?

Yeah, let's do that. That way we'll get back to those open questions and actually answer them. Good idea!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (JAORMX) [reply .github/PULL_REQUEST_TEMPLATE.md]: The reasoning makes sense. And yeah, if the user story is already referenced in the GH issue, then definitely it's not needed here in the PR template. remove it.

**Review by saschagrunert — APPROVED** [review]
/lgtm
/hold for others to review and lift

**Review by jhrozek — APPROVED** [review]
/lgtm
/hold cancel

**Comment by JAORMX** [comment]
/hold

Let's keep the hold so @cmurphy and @pjbgf can still chime in

**Review by cmurphy — APPROVED** [review]

**Review by pjbgf — APPROVED** [review]
/lgtm

**Pull request merged**
